File: 06Adapter/Students/2018/ZwaraPiotr/Adapter/Adapter.Zad1/Implementation/Seabird.cs
using Adapter.Zad1.Abstraction;

namespace Adapter.Zad1.Implementation
{
    public class Seabird : Seacraft, IAircraft
    {
        public void TakeOff()
        {
            while (!this.Airborne)
                this.IncreaseRevs();
        }

        public int Height { get; set; } // private set; }

        public override void IncreaseRevs()
        {
            base.IncreaseRevs();
            if (this.Speed > 40)
                this.Height += 100;
        }

        public bool Airborne
        {
            get { return this.Height > 50; }
        }
    }
}

File: 06Adapter/Students/2019/SebastianPiekarek/src/ZAD1/Seabird.java
package ZAD1;

public class Seabird extends Seacraft implements IAircraft{

    private int height;
    private boolean airbone;

    public Seabird(){
        this.height = 0;
    }

    @Override
    public void takeOff() {
        while (!airbone)
            IncreaseRevs();
    }

    @Override
    public void IncreaseRevs() {
        super.IncreaseRevs();
        if(speed > 40) {
            height += 100;
            this.airbone = getAirbone();
        }
    }

    public boolean getAirbone() {
        return height > 50;
    }

    public int getHeight() {
        return height;
    }



}
